dice_ml/explainer_interfaces/dice_tensorflow2.py

```diff
@@ -378,8 +378,8 @@ def stop_loop(self, itr, loss_diff):
         """Determines the stopping condition for gradient descent."""
 
         # intermediate projections
-        if((self.project_iter > 0) & (itr > 0)):
-            if((itr % self.project_iter) == 0):
+        if self.project_iter > 0 and itr > 0:
+            if (itr % self.project_iter) == 0:
```

Comment:
Probably you could drop these parenthesis as well. See the operator precedence order in the list at the link https://data-flair.training/blogs/python-operator-precedence/#:~:text=Python%20Operator%20Precedence%20%E2%80%93%20Short%20Circuiting%20Python%20always,the%20second%20operand%20only%20until%20it%20is%20needed.